File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main()
{
    string text = get_string("Text:");
    int letters = 0;
    int words = 1;   //additional word for every number of spaces
    int sentences = 0;

    //run through and compare every array individually
    for (int i = 0; i < strlen(text); i++)

    {
        if (isalpha(text[i])) //true means alphabet, no need additional condition
        {
            letters++;
        }

        if (text[i] == ' ')
        {
            words++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    int index = round(0.0588 * 100 * (float)letters / words - 0.296 * 100 * (float)sentences / words - 15.8);


    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }

}

File: pset6/readability.py
import cs50

# Text input
text = cs50.get_string("Text: ")

# Initialise variables
letters = 0
words = 1  # additional word for every number of space
sentences = 0

for i in range(len(text)):
    if text[i].isalpha():
        letters += 1

    if text[i] == " ":
        words += 1

    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

# Apply formula
L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# Print grades
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")